pkg/proxy/winkernel/proxier.go

```diff
@@ -1270,6 +1283,24 @@ func (proxier *Proxier) syncProxyRules() {
 			}
 			lbIngressIP.hnsID = hnsLoadBalancer.hnsID
 			klog.V(3).InfoS("Hns LoadBalancer resource created for loadBalancer Ingress resources", "lbIngressIP", lbIngressIP)
+
+			if gceGatewayEndpoint != nil {
+				hnsHealthCheckLoadBalancer, err := hns.getLoadBalancer(
+					[]endpointsInfo{*gceGatewayEndpoint},
+					loadBalancerFlags{isDSR: svcInfo.preserveDIP || proxier.isDSR, useMUX: svcInfo.preserveDIP, preserveDIP: svcInfo.preserveDIP},
```

Comment:
Should the flags here be synced with the flags above?

Reply:
Actually I am not sure what this annotation means here: `preserveDIP := service.Annotations["preserve-destination"] == "true"`, they are copied over.

The rest is not necessarily synced, because the only targeting client is GCLB health check, the requirement for the service itself doesn't apply to health check.